docs/DeveloperGuide.adoc

```diff
@@ -767,16 +809,222 @@ Use case ends.
+[none]
+** 2b3a. The given index is invalid.
++
+[none]
+*** 2b3a1. AddressBook shows an error message.
++
+Use case resumes at step 2b2.
+[none]
+** 2b3b. The given data is invalid.
++
+[none]
+*** 2b3b1. AddressBook shows an error message.
++
+Use case resumes at step 2b2.
+* 2b4. AddressBook edits the person with the new data
++
+Use case ends.
+
+
+[discrete]
+=== Use case: Autocomplete commands
+
+*MSS*
+
+1.  User hits an alphabet key
+2.  AddressBook drops down a list of possible commands
+3.  User requests to select a specific command in the list
+4.  AddressBook appends the remainder of the command into the input box
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User continues to type.
++
+Use case resumes at step 2.
+[none]
+* 2b. AddressBook does not display any possible command
++
+Use case ends.
+
+[none]
+* 3a. User requests to execute the command.
++
+Use case ends.
+
+[discrete]
+=== Use case: Delete tag
+
+*MSS*
+
+1. User requests to delete tag
+2. Rolodex shows a list of tags
+3. User requests to delete a specific tag in the list
+4. Rolodex deletes the tag
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. Rolodex shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Logging into the AddressBook with login credentials
+
+*MSS*
+
+1.  User opens the application
+2.  AddressBook shows a prompt for the login credentials
+3.  User requests to login with valid credentials
+4.  AddressBook initializes with the user's data
++
+Use case ends.
+
+*Extensions*
+
+* 2a. User does not have an account
+[none]
+** 2a1. User requests to create a new account
+[none]
+*** 2a1a. AddressBook creates a new account
++
+Use case resumes at step 2.
+[none]
+*** 2a1b. AddressBook failed to create a new account
++
+Use case resumes at step 2a.
+
+[none]
+* 2b. User forgot his login credentials
+[none]
+** 2b1. User requests to display the hint for help
+[none]
+*** 2b1a. User remembers the correct credentials
++
+Use case resumes at step 3
+[none]
+*** 2b1b. User cannot remember the correct credentials
+[none]
+**** 2b1ba. User calls support for help
++
+Use case ends.
+
+* 3a. The given login credentials is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Every command should respond within two seconds.
+.  Features of the product should be added incrementally.
```

Comment:
this has more to do with planning rather than NFR

Reply:
@jo-lyn